src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -46,6 +67,53 @@ function isValidFileType(fileName) {
   return FILE_TYPES.some(type => fileName.toLowerCase().endsWith(type));
 }
 
+function getBoundaryEdgeWarningDirection({ topBoundaryMet, bottomBoundaryMet, rightBoundaryMet, leftBoundaryMet }) {
+  if (topBoundaryMet) {
+    return 'down';
+  }
+
+  if (bottomBoundaryMet) {
+    return 'up';
+  }
+
+  if (leftBoundaryMet) {
+    return 'left';
+  }
+
+  if (rightBoundaryMet) {
+    return 'right';
+  }
+
+  return '';
+}
+
+function makeBoundaryEdgeWarning(direction) {
+  return `You have reached the edge of your photo and cannot move it any farther ${direction}. Use the other arrows or the make larger and make smaller buttons to continue to edit.`;
+}
+
+function hasParallelBoundariesMet({ topBoundaryMet, bottomBoundaryMet, leftBoundaryMet, rightBoundaryMet }) {
+  return (topBoundaryMet && bottomBoundaryMet) || (leftBoundaryMet && rightBoundaryMet);
+}
+
+function makeCropBoundaryWarningMessage({ zoomWarn = false, ...boundariesMet }) {
```

Comment:
Got a cognitive complexity warning on this function. This should be a lot simpler to understand.